clang/test/CIR/CodeGen/atomic-thread-fence.c

```diff
@@ -0,0 +1,149 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --check-prefix=CIR --input-file=%t.cir %s
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --check-prefix=LLVM --input-file=%t.ll %s
+
+
+struct Data {
+  int value;
+  void *ptr;
+};
+
+typedef struct Data *DataPtr;
+
+void applyThreadFence() {
+  __atomic_thread_fence(__ATOMIC_SEQ_CST);
+}
+
+// CIR-LABEL: cir.func no_proto @applyThreadFence
+// CIR:   %0 = cir.const #cir.int<5> : !s32i
+// CIR:   cir.atomic.fence(sync_scope = system, ordering = seq_cst)
+// CIR:   cir.return
+
+// LLVM-LABEL: @applyThreadFence
+// LLVM:    fence seq_cst
+// LLVM:    ret void
+
+void applySignalFence() {
+  __atomic_signal_fence(__ATOMIC_SEQ_CST);
+}
+// CIR-LABEL: cir.func no_proto @applySignalFence
+// CIR:    %0 = cir.const #cir.int<5> : !s32i
+// CIR:    cir.atomic.fence(sync_scope = singlethread, ordering = seq_cst)
+// CIR:    cir.return
+
+// LLVM-LABEL: @applySignalFence
+// LLVM:    fence syncscope("singlethread") seq_cst
+// LLVM:    ret void
+
+void modifyWithThreadFence(DataPtr d) {
+  __atomic_thread_fence(__ATOMIC_SEQ_CST);
+  d->value = 42;
+}
+// CIR-LABEL: cir.func @modifyWithThreadFence
+// CIR:    %[[DATA:.*]] = cir.alloca !cir.ptr<!ty_Data>, !cir.ptr<!cir.ptr<!ty_Data>>, ["d", init] {alignment = 8 : i64}
+// CIR:    cir.store %arg0, %[[DATA]] : !cir.ptr<!ty_Data>, !cir.ptr<!cir.ptr<!ty_Data>>
+// CIR:    %1 = cir.const #cir.int<5> : !s32i
```

Comment:
all SSA values need either not appear (not relevant for testing) or be regexed. Here and everywhere else in the PR.